libs/util/shared/src/lib/utils/temporal-extent-union.spec.ts

```diff
@@ -1,7 +1,30 @@
 import { getTemporalRangeUnion } from './temporal-extent-union'
+import { DateService } from '../services'
+
+// Create a minimal DateService mock by casting
+const dateServiceMock = {
+  formatDate: (
+    date: Date | string,
+    options?: Intl.DateTimeFormatOptions
+  ): string => {
+    const dateObj = typeof date === 'string' ? new Date(date) : date
+    return options
+      ? dateObj.toLocaleDateString('en', options)
+      : dateObj.toLocaleDateString('en')
+  },
+  formatDateTime: (
+    date: Date | string,
+    options?: Intl.DateTimeFormatOptions
+  ): string => {
+    const dateObj = typeof date === 'string' ? new Date(date) : date
+    return options
+      ? dateObj.toLocaleString('en', options)
+      : dateObj.toLocaleString('en')
+  },
+} as unknown as DateService
 
-// lock locale to en
 beforeAll(() => {
+  // Override the native toLocaleDateString to use 'en' locale if called as a fallback.
```

Comment:
Can't we set the local in a different way? this would prevent mocking and redefining DateService each time we need to test dates maybe?